server/index.js

```diff
@@ -1,144 +1,114 @@
+      },
+      validate: isNumber
+    },
+    key: {
+      cli: {
+        description: 'Path to SSL key'
+      }
+    },
+    cert: {
+      cli: {
+        description: 'Path to SSL certificate'
+      }
+    },
+    supports: {
+      cli: {
+        description: 'Range of supported client subprotocols'
+      }
+    },
+    subprotocol: {
+      cli: {
+        description: 'Server subprotocol'
+      }
+    },
+    logger: {
+      cli: {
+        alias: 'l',
+        description: 'Logger type'
+      },
+      validate: value => oneOf(['human', 'json'], value)
+    },
+    backend: {
+      cli: {
+        description: 'Backend to process actions and authentication'
+      }
+    },
+    controlSecret: {
+      cli: {
+        description: 'Secret to control Logux server'
+      }
+    },
+    controlMask: {
+      cli: {
+        description: 'CIDR masks for IP addresses of control servers'
+      }
+    },
+    redis: {
+      cli: {
+        description: 'URL to Redis for Logux Server Pro scaling'
+      }
+    },
+    ping: {},
+    root: {},
+    store: {},
+    server: {},
+    env: {},
+    fileUrl: {}
+  },
+  envPrefix: 'LOGUX',
+  examples: ['$0 --port 31337 --host 127.0.0.1', 'LOGUX_PORT=1337 $0']
 }
 
-let yargs = buildYargs()
-
-yargs
-  .option('host', {
-    alias: 'h',
-    describe: 'Host to bind server.',
-    type: 'string'
-  })
-  .option('port', {
-    alias: 'p',
-    describe: 'Port to bind server',
-    type: 'number'
-  })
-  .option('key', {
-    describe: 'Path to SSL key ',
-    type: 'string'
-  })
-  .option('cert', {
-    describe: 'Path to SSL certificate',
-    type: 'string'
-  })
-  .option('supports', {
-    describe: 'Range of supported client subprotocols',
-    type: 'string'
-  })
-  .option('subprotocol', {
-    describe: 'Server subprotocol',
-    type: 'string'
-  })
-  .option('logger', {
-    alias: 'l',
-    describe: 'Logger type',
-    choices: ['human', 'json'],
-    type: 'string'
-  })
-  .option('backend', {
-    describe: 'Backend to process actions and authentication',
-    type: 'string'
-  })
-  .option('control-secret', {
-    describe: 'Secret to control Logux server',
-    type: 'string'
-  })
-  .option('control-mask', {
-    describe: 'CIDR masks for IP addresses of control servers',
-    type: 'string'
-  })
-  .option('redis', {
-    describe: 'URL to Redis for Logux Server Pro scaling',
-    type: 'string'
-  })
-  .epilog(`Environment variables: \n${envHelp()}`)
-  .example('$0 --port 31337 --host 127.0.0.1')
-  .example('LOGUX_PORT=1337 $0')
-  .locale('en')
-  .help()
-  .version(false)
-
 export class Server extends BaseServer {
   static loadOptions (process, defaults) {
-    if (defaults.root) {
-      dotenv.config({ path: join(defaults.root, '.env') })
-    } else {
-      dotenv.config()
-    }
-    let argv = yargs.parse(process.argv)
-    let opts = {}
-
-    for (let name of AVAILABLE_OPTIONS) {
-      if (name !== 'fileUrl') {
```

Comment:
Why fileUrl has special treatment?

Reply:
`fileUrl` has no CLI argument or environment variable, it is a constructor option only for the case of ESM application (to avoid conversation `import.meta.url` to `__dirname`).

Reply:
Ok, in my design it's in JS api only